### PR TITLE
object/put: Check payload size of the finished objects

### LINKED ISSUE #580: Max size of finished objects is not checked
NeoFS network configuration dictates a limit on the maximum size of the payload of objects. Storage nodes should check if incoming objects satisfy to the limit.

In current implementation storage nodes don't check payload size of "ready" objects. As a result, excessive consumption of node memory is possible, since it tries not to cache more than the maximum value in the memory of the payload within one stream of the object (`object.Put`). 

P.S. During `object.Get` this kind of check may not be legitimate, since the constraint is a dynamic configuration.

### How to reproduce
Prepare the object with payload larger than limit and stream it w/o session.

### PR DESCRIPTION
**Pull request by cthulhu-rider**
Closes #580.

### CODE REVIEW

File: pkg/services/object/put/validation.go [hunk 1]
```diff
@@ -20,9 +21,34 @@ type validatingTarget struct {
...
+	maxPayloadSz uint64 // network config
+
+	payloadSz uint64 // payload size of the streaming object from header
+
+	writtenPayload uint64 // number of already written payload bytes
 }
 
+// errors related to invalid payload size
+var (
+	errExceedingMaxSize = errors.New("payload size is greater than the limit ")
```
Comment (alexvanin) [inline pkg/services/object/put/validation.go:34]:
```suggestion
	errExceedingMaxSize = errors.New("payload size is greater than the limit")
```

File: pkg/services/object/put/validation.go [hunk 2]
```diff
@@ -20,9 +21,34 @@ type validatingTarget struct {
...
+	t.payloadSz = obj.PayloadSize()
+	chunkLn := uint64(len(obj.Payload()))
+
+	// check chunk size
+	if chunkLn > t.payloadSz {
+		return errWrongPayloadSize
+	}
+
+	// check payload size limit
+	if t.payloadSz > t.maxPayloadSz {
```
Comment (alexvanin) [inline pkg/services/object/put/validation.go:48]: Wouldn't it better to do this?
```suggestion
	if chunkLn > t.maxPayloadSz {
```

As far as I understand, client can send 2GiB of payload in stream with correct object size in header and this check will fail, or am I missing something?

Comment (cthulhu-rider) [reply pkg/services/object/put/validation.go:48]: `chunkLn` should not be GE `payloadSz` by condition above. This condition checks that the client does not start streaming a obviously larger object. Your condition is satisfied automatically by the transitivity of the `>` operator.

File: pkg/services/object/put/validation.go (hunk 1 above)
Comment (cthulhu-rider) [reply pkg/services/object/put/validation.go:34]: Fixed thx

File: pkg/services/object/put/validation.go (hunk 2 above)
Comment (alexvanin) [reply pkg/services/object/put/validation.go:48]: `validatingTarget.WriteHeader` is not executed against unfinished objects, I missed that. Therefore my comment is invalid.

**Review by alexvanin — APPROVED** [review]

**Pull request merged**